table.go

```diff
@@ -66,39 +45,19 @@ type RoutingTable struct {
 	cplRefreshLk   sync.RWMutex
 	cplRefreshedAt map[uint]time.Time
 
-	// replacement candidates for a Cpl
-	cplReplacementCache *cplReplacementCache
-
 	// notification functions
 	PeerRemoved func(peer.ID)
 	PeerAdded   func(peer.ID)
 
-	// is peer replacement enabled ?
-	isReplaceEnabled bool
-	// peerReplaceCh is the channel to write a peer replacement request to
-	peerReplaceCh chan peer.ID
-
-	// function to determine the validity of a peer for RT membership
-	peerValidationFnc PeerValidationFunc
-	// timeout for a single call to the peer validation function
-	peerValidationTimeout time.Duration
-
-	// interval between two runs of the table cleanup routine
-	tableCleanupInterval time.Duration
-	// function to select peers that need to be validated during cleanup
-	peersForValidationFnc PeerSelectionFunc
+	// maxLastSuccessfulOutboundThreshold is the max threshold/upper limit for the value of "lastSuccessfulOutboundQuery"
+	// of the peer in the bucket above which we will evict it to make place for a new peer if the bucket
+	// is full
+	maxLastSuccessfulOutboundThreshold float64
 }
 
 // NewRoutingTable creates a new routing table with a given bucketsize, local ID, and latency tolerance.
 // Passing a nil PeerValidationFunc disables periodic table cleanup.
-func NewRoutingTable(bucketsize int, localID ID, latency time.Duration, m peerstore.Metrics,
-	opts ...Option) (*RoutingTable, error) {
-
-	var cfg options
-	if err := cfg.apply(append([]Option{Defaults}, opts...)...); err != nil {
-		return nil, err
-	}
-
+func NewRoutingTable(bucketsize int, localID ID, latency time.Duration, m peerstore.Metrics, maxLastSuccessfulOutboundThreshold float64) (*RoutingTable, error) {
```

Comment:
@aschmahmann I don't see the benefit of using Options here as other than `latency`, none of these values have meaningful defaults. What good would it do to us to have all that extra code ?

Reply:
@aarshkshah1992 your call. I find that every time we have an object missing a context or functional options we end up regretting it later since we have to break the function when we add/remove parameters.

Given that this package is really only used from go-libp2p-kad-dht to the point we've considered just merging them you're right that's probably fine for us to just change the function declaration later.